wgpu/src/lib.rs

```diff
@@ -260,6 +298,93 @@ impl MapContext {
 /// [`DeviceExt::create_buffer_init`](util::DeviceExt::create_buffer_init).
 ///
 /// Corresponds to [WebGPU `GPUBuffer`](https://gpuweb.github.io/gpuweb/#buffer-interface).
+///
+/// # Mapping buffers
+///
+/// If a `Buffer` is created with the appropriate [`usage`], it can be *mapped*:
+/// you can make its contents accessible to the CPU as an ordinary `&[u8]` or
+/// `&mut [u8]` slice of bytes. Buffers created with the
+/// [`mapped_at_creation`][mac] flag set are also mapped initially.
+///
+/// Depending on the hardware, the buffer could be memory shared between CPU and
```

Comment:
I don't actually know anything about this. Firefox doesn't use `wgpu` proper, only `wgpu-core`, and our mapping is cross-process, so there's *tons* of copies.

Since I don't understand the issues you're describing, I'd like to leave this point for someone else to explain.

Reply:
Oh, wait, now I know what you mean.